tests/test_bespoke.py

```diff
@@ -983,3 +913,229 @@ def test_bespoke_aep_is_zero_if_no_turbines_placed():
+        res_fp = os.path.join(td, 'ri_100_wtk_{}.h5')
+        excl_fp = os.path.join(td, 'ri_exclusions.h5')
+        shutil.copy(EXCL, excl_fp)
+        shutil.copy(RES.format(2013), res_fp.format(2013))
+
+        res_fp_2013 = res_fp.format('2013')
+
+        # gids 33 and 35 are included, 37 is fully excluded
+        points = pd.DataFrame({'gid': [33], 'config': ['default'],
+                               'extra_unused_data': [42]})
+
+        gid_map = pd.DataFrame({'gid': [3, 4, 13, 12, 11, 10, 9]})
+        new_gid = 50
+        gid_map['gid_map'] = new_gid
+        fp_gid_map = os.path.join(td, 'gid_map.csv')
+        gid_map.to_csv(fp_gid_map)
+
+        TechMapping.run(excl_fp, RES.format(2013), dset=TM_DSET, max_workers=1)
+
+        assert not os.path.exists(out_fpath1)
+        assert not os.path.exists(out_fpath2)
+
+        _ = BespokeWindPlants.run(excl_fp, res_fp_2013, TM_DSET,
+                                  OBJECTIVE_FUNCTION, CAP_COST_FUN,
+                                  FOC_FUN, VOC_FUN,
+                                  points, SAM_CONFIGS,
+                                  ga_kwargs={'max_time': 1},
+                                  excl_dict=EXCL_DICT,
+                                  output_request=output_request,
+                                  max_workers=1,
+                                  out_fpath=out_fpath1)
+
+        assert os.path.exists(out_fpath1)
+        assert not os.path.exists(out_fpath2)
+
+        _ = BespokeWindPlants.run(excl_fp, res_fp_2013, TM_DSET,
+                                  OBJECTIVE_FUNCTION, CAP_COST_FUN,
+                                  FOC_FUN, VOC_FUN,
+                                  points, SAM_CONFIGS,
+                                  ga_kwargs={'max_time': 1},
+                                  excl_dict=EXCL_DICT,
+                                  output_request=output_request,
+                                  max_workers=1,
+                                  out_fpath=out_fpath2,
+                                  gid_map=fp_gid_map)
+        assert os.path.exists(out_fpath2)
+
+        with Resource(out_fpath1) as f1:
+            meta1 = f1.meta
+            data1 = {k: f1[k] for k in f1.dsets}
+
+        with Resource(out_fpath2) as f2:
+            meta2 = f2.meta
+            data2 = {k: f2[k] for k in f2.dsets}
+
+        hh = SAM_CONFIGS['default']['wind_turbine_hub_ht']
+        with Resource(res_fp_2013) as f3:
+            ws = f3[f'windspeed_{hh}m', :, new_gid]
+
+        cols = ['n_gids', 'gid_counts', 'res_gids']
+        pd.testing.assert_frame_equal(meta1[cols], meta2[cols])
+
+        assert not np.allclose(data1['cf_mean-2013'], data2['cf_mean-2013'])
+        assert not np.allclose(data1['ws_mean'], data2['ws_mean'], atol=0.2)
+        assert np.allclose(ws.mean(), data2['ws_mean'], atol=0.01)
+
+
+if __name__ == '__main__':
+    output_request = ('system_capacity', 'cf_mean', 'cf_profile',
```

Comment:
This is the bias correct test, right? Should it be in it's own test function or is this intentional?